### PR TITLE
restore: Fix system tables are filtered bug

### PR DESCRIPTION
**Pull request by zwj-coder**
### What problem does this PR solve? 
Fix issue [#1197](https://github.com/pingcap/br/issues/1197) and [#1201](https://github.com/pingcap/br/issues/1201)
When user only want to restore the system table, i.e. -f "mysql*.*", because the system database was rename to a temporary name, that name will be filtered causing no system table is restored 

### What is changed and how it works?
When filter restore databases and tables, revert the temporary name to its original name 

### Check List 

Tests 

 - Integration test
 
Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### Release note

 - No Release Note

### CODE REVIEW

File: tests/br_systables/run.sh [hunk 1]
```diff
@@ -47,8 +65,28 @@ check() {
...
+    run_sql "SELECT user FROM mysql.user WHERE user='newuser';" | grep 'newuser'
+}
+
 modify_systables
 run_br backup full -s "local://$backup_dir"
 rollback_modify
 run_br restore full -f '*.*' -f '!mysql.bar' -s "local://$backup_dir"
 check
+
+run_br restore full -f 'mysql.bar' -s "local://$backup_dir"
```
Comment (3pointer) [inline tests/br_systables/run.sh:79]: better to check `nothing to restore` here 🤔️?

File: pkg/utils/schema.go [hunk 2]
```diff
@@ -104,3 +104,10 @@ func IsSysDB(dbLowerName string) bool {
 func TemporaryDBName(db string) model.CIStr {
 	return model.NewCIStr("__TiDB_BR_Temporary_" + db)
 }
+
+// GetSysDBName get the original name of system DB
+func GetSysDBName(tempDB model.CIStr) (string, bool) {
+	ok := strings.Contains(tempDB.O, "__TiDB_BR_Temporary_")
+	i := strings.LastIndex(tempDB.O, "_")
+	return tempDB.O[i+1:], ok
```
Comment (kennytm) [inline pkg/utils/schema.go:112]: it should be checking if `ok := strings.HasPrefix(tempDB.O, "__TiDB_BR_Temporary_")`, right? And the `i+1` is the length of that temporary string.

Comment (zwj-coder) [reply pkg/utils/schema.go]: fixed this method

**Review by 3pointer — APPROVED** [review]

**Review by kennytm — APPROVED** [review]

**Pull request merged**